internal/cmd/service.go

```diff
@@ -0,0 +1,147 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/brasic/launchd"
+	"github.com/brasic/launchd/state"
+	"github.com/spf13/cobra"
+)
+
+func newServiceCmd(ctx context.Context, logger *log.Logger) *cobra.Command {
+	short := "Manage this program as a launchd system service."
+	svc := rdmService()
+	cmd := &cobra.Command{
+		Use:   "service [subcommand]",
+		Short: short,
+		Long:  short + "\n" + prettyStatus(svc),
+	}
+	cmd.AddCommand(serviceInstallCmd(ctx, logger))
+	cmd.AddCommand(serviceUninstallCmd(ctx, logger))
+	cmd.AddCommand(serviceStartCmd(ctx, logger))
+	cmd.AddCommand(serviceStopCmd(ctx, logger))
+	return cmd
+}
+
+func serviceInstallCmd(ctx context.Context, logger *log.Logger) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "install",
+		Short: "Configures rdm to run on boot as a MacOS LaunchAgent.",
+		Run: func(cmd *cobra.Command, args []string) {
+			svc := rdmService()
+
+			if svc.IsHealthy() {
+				fmt.Println("service is already installed and running, nothing to do!")
+				return
+			}
+			// Configure launchagent to run `rdm server` at boot
+			if err := svc.Install(); err != nil {
+				fmt.Printf("Problem installing: %v\n", err)
+				return
+			}
+
+			// In case the service is not running, start it.
+			if err := svc.Start(); err != nil {
+				fmt.Printf("Problem starting: %v\n", err)
+				return
+			}
+
+			fmt.Printf("Configured to start at boot. Uninstall using:\n\t%s service uninstall\n", currentExecutableName())
```

Comment:
Should these `Println`s use the logger? The logger should only be file backed for the `server` command.

Reply:
I don't quite grok this part of your question:

> The logger should only be file backed for the server command.

That said, the `service` commands are always intended to be invoked by humans and these strings are for interactive presentation at the time they invoke the command.  Can you help me understand why a log line would be better here?

Reply:
That comment had some built-in assumed context, sorry about that.

I've used the logger to output to STDOUT in most places iirc, except for `server` so there's a consistent interface/place to log. That way it's easy to move how/where things are logged to without making a ton of sweeping changes. e.g. if we wanted a silent version of this command.

If there's a meaningful difference between logging via `fmt.Printf` and the `logger`, feel free to ignore me. 👍

Reply:
The meaningful difference is this one:

```
./remote-development-manager service stop
2022/11/12 10:23:31 Service is not running, nothing to do.
```

vs

```
./remote-development-manager service stop
Service is not running, nothing to do.
```

From an aesthetic POV if I run a command and see a message meant for me to read immediately I would think something was broken or wrong if it looked like a log entry.

This is definitely nitpicking and bikeshedding but there's a difference between "event messages" from a running daemon in a log that could happen at any time and record when a thing happened and "response messages" from a command which only happen as a direct response to user input to let them know the result of their action.


WDYT about one of the following two options?

 1. passing a separate "interactive" logger configured without timestamps to cobra command objects when invoked from the CLI? Then we could use the logger interface here without the ugly timestamps, and obtain the ability to use these things in a headless context in the future by passing them a different logger.
 2. Swapping most of these messages that are really just "hey, you asked me to start this service and I did so I'm going tell you to be nice" to use stderr instead.
 3.

Reply:
> This is definitely nitpicking and bikeshedding but there's a difference between "event messages" from a running daemon in a log that could happen at any time and record when a thing happened and "response messages" from a command which only happen as a direct response to user input to let them know the result of their action.

Good point, I agree with that.

> * passing a separate "interactive" logger configured without timestamps to cobra command objects when invoked from the CLI? Then we could use the logger interface here without the ugly timestamps, and obtain the ability to use these things in a headless context in the future by passing them a different logger.

No strong feelings on which approach we take (feel free to use your best judgement), but the advantage of being able to log the information no matter the context (headless or interactive, etc.) makes me lean slightly towards this one.

Reply:
I feel pretty good about the approach I took in https://github.com/BlakeWilliams/remote-development-manager/pull/16/commits/5c6ea00f5b4e442d4922099a64202f423d6780d6.  In the process I found that there were a number of places where the default stderr logger was being used instead of the server logger field.